src/NzbDrone.Common.Test/Http/HttpUriFixture.cs

```diff
@@ -0,0 +1,84 @@
+﻿using FluentAssertions;
+using NUnit.Framework;
+using NzbDrone.Common.Http;
+using NzbDrone.Test.Common;
+
+namespace NzbDrone.Common.Test.Http
+{
+    public class HttpUriFixture : TestBase
+    {
+        [TestCase("", "", "")]
+        [TestCase("/", "", "/")]
+        [TestCase("base", "", "base")]
+        [TestCase("/base", "", "/base")]
+        [TestCase("/base/", "", "/base/")]
+        [TestCase("", "relative", "relative")]
+        [TestCase("", "/relative", "/relative")]
+        [TestCase("/", "relative", "/relative")]
+        [TestCase("/", "/relative", "/relative")]
+        [TestCase("base", "relative", "relative")]
+        [TestCase("base", "/relative", "/relative")]
+        [TestCase("/base", "relative", "/relative")]
```

Comment:
Huh, when you explain it like that, it makes a lot of sense. For the URL Base issue we should be requiring a trailing slash then?